### PR TITLE
Change client-side credentials' use of gRPC Core

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
This depends upon [pull request 13539](https://github.com/grpc/grpc/pull/13593) and is forward progress on [issue 12531](https://github.com/grpc/grpc/issues/12531).

### CODE REVIEW

File: src/python/grpcio/grpc/_cython/_cygrpc/credentials.pxd.pxi [hunk 1]
```diff
@@ -12,20 +12,66 @@
...
+
+  cdef grpc_call_credentials *c(self)
 
 
 cdef class ChannelCredentials:
 
+  cdef grpc_channel_credentials *c(self)
+
+  # TODO(https://github.com/grpc/grpc/issues/12531): remove.
   cdef grpc_channel_credentials *c_credentials
```
Comment (kpayson64) [inline src/python/grpcio/grpc/_cython/_cygrpc/credentials.pxd.pxi]: Why is this still needed?

It looks like it doesn't get set by any of its subclasses.  (Same with ```CallCredentials.c_credentials```)

File: src/python/grpcio/grpc/__init__.py [hunk 2]
```diff
@@ -401,14 +400,14 @@ class AuthMetadataPlugin(six.with_metaclass(abc.ABCMeta)):
     def __call__(self, context, callback):
         """Implements authentication by passing metadata to a callback.
 
-    Implementations of this method must not block.
+        Implementations of this method must not block.
 
-    Args:
-      context: An AuthMetadataContext providing information on the RPC that the
-        plugin is being called to authenticate.
-      callback: An AuthMetadataPluginCallback to be invoked either synchronously
```
[the remark this answers is not in the document]
Comment (nathanielmanistaatgoogle) [reply src/python/grpcio/grpc/__init__.py]: Note that within the Cython code we introduce asynchronicity in calling the callback created in `_plugin_wrapping`, so the application's synchronously calling the callback that `_plugin_wrapping` passes to it counts as asynchronous with respect to gRPC Core.

(We may want to revisit this in the future with an aim toward eliminating one more randomly spawned thread, but not in this change at this time.)

File: src/python/grpcio/grpc/_cython/_cygrpc/credentials.pxd.pxi (hunk 1 above)
Comment (nathanielmanistaatgoogle) [reply src/python/grpcio/grpc/_cython/_cygrpc/credentials.pxd.pxi]: Google-internal code for the moment still depends upon the existence of this field; after a two-repository-four-change funky dance routine the field should be eliminable. :-)

**Comment by nathanielmanistaatgoogle** [comment]
(Not sure what's up with GitHub but all tests have passed; submitting now.)

**Pull request merged**